File: problems/leetcode/1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> c;

        for (int i = 0; i<nums.size(); i++) {
            if (c.contains(target - nums[i])) {
                return {c.at(target - nums[i]), i};
            }
            c[nums[i]] = i;
        }
        return {0, 0};
    }
};

File: problems/leetcode/1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        diff = {} 
        for i in range(len(nums)):
            missing = target-nums[i]
            if missing in diff.keys():
                return [diff[missing][1],i]
            diff[nums[i]] = [missing,i]
        return diff